### ISSUE TITLE
Proposal: Adjust branching/release strategy post-terra 2

### ISSUE THREAD

**Issue by drewboswell** [terra-money/mantlemint#50]
### TLDR; 
Make terra v2 main, classic a maintenance branch.
Create deterministic tagged releases.
Reduce number of branches to maintain and track.

### Current
perceived logic: permanent branch by chain-id, no releases for now, track head on branch.
* branch/main -> terra classic
* branch/columbus-5 -> terra classic mainnet (same as main)
* branch/pisco-1 -> terra v2 testnet
* branch/phoenix-1 -> terra v2 mainnet (two line dependency diff from pisco-1)

### Proposal
logic: github tag+releases done from main and maintenance branches only. Active development comes back to main else it is a maintenance release.

* branch: `main` --> terra v2
  * permanent main development branch, terra v2
  * release major version: 1.x.x
    * use semver versioning
    * examples: `1.0.0-rc.0`, `1.0.0`
* branch: `classic` -> terra-classic maintenance branch
  * permanent terra-classic maintenance branch
  * release major version: 0.x.x
    * use semver versioning
    * examples: `0.1.5`, `0.1.5-1`, `0.1.5-rc.0`, `0.1.5-2`
* branch:  `feat/largetask` --> temporary feature branch needing longer than one signle PR, to be merged to main
  * no releases
  * example: previous`upgrade/terra2` branch

### Example migration
* merge branch `columbus-5` into `main`
* create branch `classic` from `main`
* merge branch `pisco-1` into `phoenix-1`
* merge branch `phoenix-1` into `main`

Remaining permanent branches: main, classic

**Comment by drewboswell** [comment]
I still think this is relevant. I'm not liking tracking head of a branch for production deployments.

The latest pisco update is a good example of this, it should be an RC release from main branch IMO.

**Comment by gregnuj** [comment]
implemented classic branch, 
goting forward we will use main for mainnet, test for testnet, and classic for classic and each release will get semantic versions coresponding to the core release (although we may need to do something slightly different for classic as there next release will be 2.0.0)
